rust/chains/hyperlane-ethereum/src/rpc_clients/fallback.rs

```diff
@@ -98,9 +171,83 @@ impl From<FallbackError> for ProviderError {
     }
 }
 
+async fn handle_stalled_provider<C>(
+    fallback_provider: &FallbackProvider<C>,
+    priority: &PrioritizedProviderInner,
+) -> Result<(), ProviderError>
+where
+    C: JsonRpcClient,
+{
+    let now = Instant::now();
+    if now
+        .duration_since(priority.last_block_height.1)
+        .le(&fallback_provider.max_block_time)
+    {
+        // Do nothing, it's too early to tell if the provider has stalled
+        return Ok(());
+    }
+
+    let provider = &fallback_provider.inner.providers[priority.index];
+    let current_block_height: u64 = provider
+        .request(BLOCK_NUMBER_RPC, ())
+        .await
+        .map(|r: U64| r.as_u64())
+        .unwrap_or(priority.last_block_height.0);
+    if current_block_height == priority.last_block_height.0 {
```

Comment:
I think it's worth expanding this case to `current_block_height <= priority.last_block_height.0`

while unlikely, we've seen some very janky stuff in the past so I could see there being a case where a provider gives a lower block number than before, which we'd also wanna deprioritize on

Providers usually have multiple nodes behind a load balancer and sometimes requests go to different nodes with different views of the world - or a node could even get in a weird state where it has a corrupted datadir and its tip gets reset to a lower block number